internal/core/src/segcore/SegmentGrowingImpl.cpp

```diff
@@ -90,6 +90,11 @@ SegmentGrowingImpl::Insert(int64_t reserved_offset,
                            InsertRecordProto* insert_record_proto) {
     AssertInfo(insert_record_proto->num_rows() == num_rows,
                "Entities_raw count not equal to insert size");
+    // protect schema being changed during insert
+    // schema change cannot happends during insertion,
+    // otherwise, there might be some data not following new schema
+    std::shared_lock lck(sch_mutex_);
```

Comment:
we should protect all read/write access of schema_, but discussed offline, we can fix 50% of the bug in this PR and fix the remaining 50% in the next one.